internal/appsec/listener/grpcsec/grpc.go

```diff
@@ -85,14 +85,21 @@ func newWafEventListener(wafHandle *waf.Handle, cfg *config.Config, limiter limi
 func (l *wafEventListener) onEvent(op *types.HandlerOperation, handlerArgs types.HandlerOperationArgs) {
 	// Limit the maximum number of security events, as a streaming RPC could
 	// receive unlimited number of messages where we could find security events
-	const maxWAFEventsPerRequest = 10
 	var (
 		nbEvents atomic.Uint32
 		logOnce  sync.Once // per request
-
-		events []any
-		mu     sync.Mutex // events mutex
 	)
+	addEvents := func(events []any) {
+		const maxWAFEventsPerRequest = 10
+		if nbEvents.Load() >= maxWAFEventsPerRequest {
+			logOnce.Do(func() {
+				log.Debug("appsec: ignoring the rpc message due to the maximum number of security events per grpc call reached")
```

Comment:
When I first read this I thought that the requests will stop being processed because there is too much appsec events. Which is not the case.

```suggestion
				log.Debug("appsec: ignoring new WAF event due to the maximum number of security events per grpc call reached")
```